examples/server/server.cpp

```diff
@@ -243,6 +243,8 @@ struct server_slot {
 
         if (params.n_predict != -1) {
             n_remaining = params.n_predict - n_decoded;
+        } else if (global_params.n_predict == -2) {
+            n_remaining = global_params.n_ctx - n_decoded;
```

Comment:
This is not precise. It should use the slot's context. And instead of `n_decoded` it has to use `n_past`. Writing a server test to verify the implementation would be useful.